src/pathfinding_igvc/test/path-finding-test.cpp

```diff
@@ -0,0 +1,219 @@
+#include <PathFinding.h>
+#include <gtest/gtest.h>
+
+TEST(PathFinding, testWeightedSum1) {
+    std::vector<float> testVec;
+    testVec.push_back(3);
+    testVec.push_back(3);
+    testVec.push_back(3);
+
+    EXPECT_EQ(5.5, PathFinding::weightedSum(testVec, 3));
+}
+
+TEST(PathFinding, testWeightedSum2) {
+    std::vector<float> testVec;
+    testVec.push_back(1);
+    testVec.push_back(2);
+    testVec.push_back(3);
+
+    EXPECT_EQ(1.0, PathFinding::weightedSum(testVec, 1));
+}
+
```

Comment:
A test for vectors with negative values might be another good case to consider here.

Reply:
Done